code/DynamicCache.php

```diff
@@ -397,14 +412,32 @@ public function run($url)
         // This is normally called in VersionedRequestFilter.
         Versioned::choose_site_stage();
 
+        // Get cache
```

Comment:
Nah it's all good. My expectation that PRing back in would be unlikely.
I don't think your solution will help me out here, so I might just make this be an Injected/customized DynamicCache object.